packages/angular/src/utils/user-installed-angular-versions.ts

```diff
@@ -0,0 +1,44 @@
+import type { Tree } from '@nrwl/devkit';
+import { readJson } from '@nrwl/devkit';
+import { clean, coerce, major } from 'semver';
+import { angularVersion } from './versions';
+
+export function getGeneratorDirectoryForInstalledAngularVersion(tree: Tree) {
+  const majorAngularVersion = getUserInstalledAngularMajorVersion(tree);
+
+  const directoryDictionary = {
+    14: 'angular-v14',
+  };
+
+  return directoryDictionary[majorAngularVersion] ?? null;
+}
+
+export function getUserInstalledAngularVersion(tree: Tree) {
+  const pkgJson = readJson(tree, 'package.json');
+  const installedAngularVersion =
+    pkgJson.dependencies && pkgJson.dependencies['@angular/core'];
+
+  if (
+    !installedAngularVersion ||
+    installedAngularVersion === 'latest' ||
+    installedAngularVersion === 'next'
+  ) {
+    return clean(angularVersion) ?? coerce(angularVersion).version;
+  }
+
+  return (
+    clean(installedAngularVersion) ?? coerce(installedAngularVersion).version
+  );
+}
+
+export function getUserInstalledAngularMajorVersion(tree: Tree): number {
+  return major(getUserInstalledAngularVersion(tree));
+}
+
+export function getUserInstalledAngularVersionMatrix(tree: Tree) {
```

Comment:
What about `getUserInstalledAngularVersionInfo`? Matrix sounds something different to me, where you have a mapping of different versions.

Reply:
I like it